server/src/sas/CodeZoneManager.ts

```diff
@@ -2457,8 +2461,11 @@ export class CodeZoneManager {
     }
   }
   private _currentZone(line: number, col: number) {
-    const newToken = this._token(line, col)!,
-      type = newToken.type; //self.type(line,col),
+    const newToken = this._token(line, col)!;
+    const type = newToken.type;
+    if (type === "embedded-code") {
```

Comment:
@scnwwu 
I found a simple solution. I add this check here, so that if it's embedded, stop here, and not reach the later startFrom() logic.
Actually the line 2464: _"const newToken = this._token(line, col)!;"_ also calls the SyntaxProvider to get the correct "embedded-code" token. If any concern, please raise a comment.